src/System.Xml.ReaderWriter/src/System/Xml/Core/XmlTextReaderImpl.cs

```diff
@@ -23,6 +23,20 @@ namespace System.Xml
 {
     internal partial class XmlTextReaderImpl : XmlReader, IXmlLineInfo, IXmlNamespaceResolver
     {
+        private static UTF8Encoding s_utf8BomThrowing;
+        
+        private static UTF8Encoding UTF8BomThrowing
+        {
+            get
+            {
+                if (s_utf8BomThrowing == null)
+                {
+                    s_utf8BomThrowing = new UTF8Encoding(encoderShouldEmitUTF8Identifier: true, throwOnInvalidBytes: true);
+                }
+                return s_utf8BomThrowing;
+            }
+        }
```

Comment:
Nit: This could be simplified as:

```csharp
private static UTF8Encoding UTF8BomThrowing =>
    s_utf8BomThrowing ?? (s_utf8BomThrowing = new UTF8Encoding(encoderShouldEmitUTF8Identifier: true, throwOnInvalidBytes: true));
```